lib/logwriter.c

```diff
@@ -1093,6 +1093,11 @@ log_writer_format_log(LogWriter *self, LogMessage *lm, GString *result)
         }
 
     }
+
+  if (self->options->truncate_size != -1 && result->len > self->options->truncate_size)
+    {
+      g_string_truncate(result, self->options->truncate_size);
```

Comment:
I wonder if we could help the user with a warning message.
Our default `log-msg-size()` is 64k, so this problem can happen when the `log-msg-size()` is increased or when using `template()`.

I know emitting warning/debug messages on each message can lower the performance, and spam the internal() log...
I'm just thinking out loud.

Reply:
maybe a stat counter could be added for tracking number of truncated messages (and maybe another counter could be useful: truncated bytes - but I think the first ctr would be more than enough)

Reply:
I would prefer the counter.

Reply:
The following counters were added (for all logwriter based destination)
```
dst.network;d_local#0;udp,127.0.0.1:1111;a;truncated_count;1
dst.network;d_local#0;udp,127.0.0.1:1111;a;truncated_bytes;1
```